SMLHelper/Patchers/CraftDataPatcher_Subnautica.cs

```diff
@@ -1,97 +1,301 @@
+                        _techType = techType,
+                        _craftAmount = smlTechData.craftAmount
+                    };
 
-                        var ingredient = new CraftData.Ingredient(smlIngredient.techType, smlIngredient.amount);
-                        ingredientsList.Add(smlIngredient.techType, smlIngredient.amount);
-                    }
-                    techDataInstance._ingredients = ingredientsList;
-                }
+                    var ingredientsList = new CraftData.Ingredients();
 
-                if (smlTechData.linkedItemCount > 0)
-                {
-                    var linkedItems = new List<TechType>();
-                    for (int l = 0; l < smlTechData.linkedItemCount; l++)
+                    if (smlTechData.ingredientCount > 0)
                     {
-                        linkedItems.Add(smlTechData.GetLinkedItem(l));
+                        for (int i = 0; i < smlTechData.ingredientCount; i++)
+                        {
+                            IIngredient smlIngredient = smlTechData.GetIngredient(i);
+
+                            var ingredient = new CraftData.Ingredient(smlIngredient.techType, smlIngredient.amount);
+                            ingredientsList.Add(smlIngredient.techType, smlIngredient.amount);
+                        }
+                        techDataInstance._ingredients = ingredientsList;
                     }
-                    techDataInstance._linkedItems = linkedItems;
-                }
 
-                bool techDataExists = CraftData.techData.ContainsKey(techType);
+                    if (smlTechData.linkedItemCount > 0)
+                    {
+                        var linkedItems = new List<TechType>();
+                        for (int l = 0; l < smlTechData.linkedItemCount; l++)
+                        {
+                            linkedItems.Add(smlTechData.GetLinkedItem(l));
+                        }
+                        techDataInstance._linkedItems = linkedItems;
+                    }
 
-                if (techDataExists)
-                {
-                    CraftData.techData.Remove(techType);
-                    Logger.Log($"{techType} TechType already existed in the CraftData.techData dictionary. Original value was replaced.", LogLevel.Warn);
-                    replaced++;
-                }
-                else
-                {
-                    added++;
+                    if (techExists)
+                    {
+                        CraftData.techData.Remove(techType);
+                        Logger.Log($"{techType} TechType already existed in the CraftData.techData dictionary. Original value was replaced.", LogLevel.Warn);
+                        replaced++;
+                        TechDataAltered++;
+                    }
+                    else
+                    {
+                        added++;
+                        TechDataAltered++;
```

Comment:
Instead of incrementing `TechDataAltered` every time you also increment `replaced` and `added`,  
you can simply set  the value once at the end of the loop.
```CS
TechDataAltered = added + replaced;
```

Reply:
I actually moved away from using that check. Just forgot to remove it. Will do